cmd/buildProviderCodeship_test.go

```diff
@@ -0,0 +1,127 @@
+
+	shipmentJSON = strings.Replace(shipmentJSON, "${name}", name, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${env}", env, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${barge}", barge, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${replicas}", strconv.Itoa(replicas), 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${group}", group, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${foo}", foo, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${property}", property, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${product}", product, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${project}", project, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${envLevel}", envLevel, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${containerLevel}", containerLevel, 1)
+	shipmentJSON = strings.Replace(shipmentJSON, "${container}", container, 1)
+	t.Log(shipmentJSON)
+
+	//deserialize shipit json
+	var shipment ShipmentEnvironment
+	err := json.Unmarshal([]byte(shipmentJSON), &shipment)
+	check(err)
+
+	//first convert shipit model to docker-compose
+	dockerCompose := transformShipmentToDockerCompose(&shipment)
+
+	//debug
+	data, _ := yaml.Marshal(dockerCompose)
+	t.Log(string(data))
+
+	//test
+	harborCompose := transformShipmentToHarborCompose(&shipment, &dockerCompose)
+
+	//debug
+	data, _ = yaml.Marshal(harborCompose)
+	t.Log(string(data))
+
+	svc := dockerCompose.Services[container]
+	assert.NotNil(t, svc)
+
+	//load circleciv1 build provider
+	provider, err := getBuildProvider("codeship")
+	if err != nil {
+		t.Fail()
+	}
+
+	//run the build provider
+	artifacts, err := provider.ProvideArtifacts(&dockerCompose, &harborCompose, "token")
+	if err != nil {
+		t.Fail()
+	}
+
+	//debug
+	data, _ = yaml.Marshal(dockerCompose)
+	t.Log(string(data))
+
+	//docker compose configuration should have the build directive
+	assert.NotEmpty(t, svc.Build)
+
+	//docker compose configuration should have the codeship build number in the image tag
+	assert.True(t, strings.HasSuffix(svc.Image, "-${CI_BUILD_ID}"))
+
+	//docker compose configuration shouldn't have any environment variables
+	assert.Equal(t, 0, len(svc.Environment))
+
+	//docker compose configuration shouldn't have any ports
+	assert.Equal(t, 0, len(svc.Ports))
+
+	assert.NotNil(t, artifacts)
+	assertArtifact(t, artifacts, "codeship-services.yml")
+	assertArtifact(t, artifacts, "codeship-steps.yml")
+	assertArtifact(t, artifacts, "codeship.env")
+	assertArtifact(t, artifacts, "codeship.aes")
+
+	//asert that docker-push.sh is writable
+	dockerPush := assertArtifact(t, artifacts, "docker-push.sh")
+	assert.True(t, dockerPush.FileMode == 0777, "expecting docker-push.sh file mode to be executable")
+
+	//assert that codeship.env and codeship.aes are added to .gitignore
+	gitignorebits, err := ioutil.ReadFile(".gitignore")
+	check(err)
+	gitignore := string(gitignorebits)
+	assert.Contains(t, gitignore, "codeship.env")
+	assert.Contains(t, gitignore, "codeship.aes")
```

Comment:
do we wan't to add to dockerignore also?

Reply:
Yes!  Great idea.